Controllers/PostController.cs

```diff
@@ -86,4 +86,35 @@ public IActionResult CreatePost(Post post)
         _dbContext.SaveChanges();
         return Created($"/api/post/{post.Id}", post);
     }
+
+    [HttpDelete("{postId}")]
+    [Authorize]
+    public IActionResult DeletePost( int postId)
+    {
+        Post PostToDelete = _dbContext.Posts.SingleOrDefault(p => p.Id == postId);
+        if (PostToDelete == null)
+        {
+            return NotFound();
+        }
+        else if (postId != PostToDelete.Id)
```

Comment:
For a delete endpoint, not sure this "else if" condition is needed. This would be necessary for a put (where it is possible for a passed-in id and a passed-in object to have conflicting info), but in this case you're just evaluating if the id corresponding to an object in the posts collection exists. It either does (handled by the final return logic) or it doesn't (which is handled by the "if null" condition). Thanks for coming to my TED talk.